core/validation/validate.py

```diff
@@ -155,6 +155,7 @@ def validate_types(
                         expected_type=exp_type,
                         actual_type=got_type,
                         row_indexes=[index],
+                        failed_row=pd.DataFrame(dict(row), index=[index]),
```

Comment:
Can we not just return the row as a series?

Reply:
The problem with this is that we need the index of the row to pass to the function that gets the cell index. We can now add row_indexes as an additional variable to work this out. I didn't want to do this before because with multiple indexes and multiples Series there is always the possibility of them getting decoupled and somehow rearranged in an incorrect order (with, for instance, some unforeseen code change down the line). 

I don't think I see any issues if we always expect both to have just one value. It will create difficulties if we move away from that model, but it looks like we're moving to streamline to just return one index per failure anyway.

Reply:
Why do we need to pass more than one index? Think will always return a single row per Failure now I think?

Reply:
Yes, now it has been rewritten that is the case, but row_indexes still implies there will be multiple indexes, and it's still an array rather than an integer, so we probably need to refactor all of the functions to return row_index as a single integer to make this clearer in the upcoming refactor of validate.py. 

Now the function has been rewritten to only return a single index I will revert failed_row back to a series